src/Umbraco.Core/Persistence/EntityNotFoundException.cs

```diff
@@ -1,38 +1,95 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Runtime.Serialization;
 
 namespace Umbraco.Core.Persistence
 {
-
     // TODO: Would be good to use this exception type anytime we cannot find an entity
 
     /// <summary>
-    /// An exception used to indicate that an umbraco entity could not be found
+    /// An exception used to indicate that an Umbraco entity could not be found.
     /// </summary>
+    /// <seealso cref="System.Exception" />
+    [Obsolete("This exception is only used in MediaController and could be replaced with an HTTP 404 error instead.")]
+    [Serializable]
     public class EntityNotFoundException : Exception
     {
+        /// <summary>
+        /// Gets the identifier.
+        /// </summary>
+        /// <value>
+        /// The identifier.
+        /// </value>
         public object Id { get; private set; }
```

Comment:
Same as in `DataOperationException`, this property can contain a type that isn't serializable.